sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/ModelBridgeInternal.java

```diff
@@ -502,36 +495,115 @@ public static void setFeedOptionsMaxItemCount(FeedOptions feedOptions, Integer m
+            return (T) c.getDeclaredConstructor(String.class).newInstance(jsonString);
         } catch (InstantiationException | IllegalAccessException | InvocationTargetException | NoSuchMethodException | IllegalArgumentException e) {
             throw new IllegalArgumentException(e);
         }
     }
 
-    public static void populatePropertyBagJsonSerializableWrapper(JsonSerializableWrapper jsonSerializableWrapper) {
-        jsonSerializableWrapper.populatePropertyBag();
-    }
-
-    public static Resource getResourceFromResourceWrapper(ResourceWrapper resourceWrapper) {
-        if (resourceWrapper == null) {
-            return null;
+    public static <T> void invokePopulatePropertyBag(T t) {
+        if (t instanceof JsonSerializable) {
+            ((JsonSerializable) t).populatePropertyBag();
+        } else if (t instanceof CompositePath) {
+            ((CompositePath) t).populatePropertyBag();
+        } else if (t instanceof ConflictResolutionPolicy) {
+            ((ConflictResolutionPolicy) t).populatePropertyBag();
+        } else if (t instanceof ConsistencyPolicy) {
+            ((ConsistencyPolicy) t).populatePropertyBag();
+        } else if (t instanceof DatabaseAccountLocation) {
+            ((DatabaseAccountLocation) t).populatePropertyBag();
+        } else if (t instanceof ExcludedPath) {
+            ((ExcludedPath) t).populatePropertyBag();
+        } else if (t instanceof IncludedPath) {
+            ((IncludedPath) t).populatePropertyBag();
+        } else if (t instanceof IndexingPolicy) {
+            ((IndexingPolicy) t).populatePropertyBag();
+        } else if (t instanceof PartitionKeyDefinition) {
+            ((PartitionKeyDefinition) t).populatePropertyBag();
+        } else if (t instanceof SpatialSpec) {
+            ((SpatialSpec) t).populatePropertyBag();
+        } else if (t instanceof SqlParameter) {
+            ((SqlParameter) t).populatePropertyBag();
+        } else if (t instanceof SqlQuerySpec) {
+            ((SqlQuerySpec) t).populatePropertyBag();
+        } else if (t instanceof UniqueKey) {
+            ((UniqueKey) t).populatePropertyBag();
+        } else if (t instanceof UniqueKeyPolicy) {
+            ((UniqueKeyPolicy) t).populatePropertyBag();
+        } else if (t instanceof Index) {
+            ((Index) t).populatePropertyBag();
+        } else if (t instanceof CosmosError) {
+            ((CosmosError) t).populatePropertyBag();
+        } else {
+            throw new IllegalArgumentException("populatePropertyBag method does not exists in class " + t.getClass());
         }
-        return resourceWrapper.getResource();
     }
 
-    public static JsonSerializable getJsonSerializableFromIndex(Index index) {
-        return index.getJsonSerializable();
+    public static <T> JsonSerializable invokeGetJsonSerializable(T t) {
```

Comment:
can `t` be null?

Reply:
It could be. 
One example is in CollectionCrudTest::deleteCollection.